clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -2365,9 +2614,21 @@ Expected<SmallVector<StringRef>> linkAndWrapDeviceFiles(
     // First link and remove all the input files containing bitcode if
     // the target linker does not support it natively.
     SmallVector<StringRef> InputFiles;
-    if (!linkerSupportsLTO(LinkerArgs))
-      if (Error Err = linkBitcodeFiles(Input, InputFiles, LinkerArgs))
+    llvm::Triple Triple(LinkerArgs.getLastArgValue(OPT_triple_EQ));
+    if (!linkerSupportsLTO(LinkerArgs)) {
+      if (sycl::isSYCLThinLTO(Args, Triple)) {
+        // For thinLTO, we consider device libs as normal compiler input
+        // and add them to the files to be processed by the LTO backend.
```

Comment:
This is something we should review with the community. As of now, they import some parts of GPU devicelib at compile time (i.e. LLVM IR). Considering community interest in moving this action to the link step, we better align with them.

Looking at some patches in upstream, I feel like some calls to GPU libraries are handled at link step (e.g. https://github.com/llvm/llvm-project/pull/98512).